### PR TITLE
Adding MODULE.bazel

### PR DESCRIPTION
**Pull request by linzhp**
This is the first step of enabling bzlmod

### CODE REVIEW

**Comment by motiejus** [comment]
Does this mean that when the customers import bazel-zig-cc via bzlMod, they will get the transitive deps (rules_go, protobuf, etc) too?

To be clear, this is *not* what we want. :)

**Comment by linzhp** [comment]
Marking all of them as `dev_dependency`, so no customer will get them via bazel-zig-cc

File: MODULE.bazel [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+bazel_dep(name = "rules_go", version = "0.38.1", dev_dependency = True)
+
+go_sdk = use_extension(
+    "@rules_go//go:extensions.bzl",
+    "go_sdk",
+    dev_dependency = True,
+)
+use_repo(go_sdk, "go_default_sdk")
+
+bazel_dep(name = "gazelle", version = "0.29.0", dev_dependency = True)
```
Comment (motiejus) [inline MODULE.bazel:15]: will this deprecate the hardcoded sha256s in WORKSPACE?

**Review by motiejus — APPROVED** [review]

File: MODULE.bazel [hunk 2]
```diff
@@ -0,0 +1,26 @@
+module(
+    name = "bazel-zig-cc",
```
Comment (motiejus) [inline MODULE.bazel:2]:
```suggestion
    name = "hermetic_cc_toolchain",
```

**Comment by aignas** [comment]
FYI, you should also add an empty `WORKSPACE.bzlmod` to ensure that `WORKSPACE` file is not used under `bzlmod`.

**Comment by motiejus** [comment]
> FYI, you should also add an empty `WORKSPACE.bzlmod` to ensure that `WORKSPACE` file is not used under `bzlmod`.

Last I spoke to @linzhp, bzlmod is not ready on this repo yet (and thus will not be ready for v2.0). He said he would do it, but we don't have a timeline.

Happy to accept contributions though.

**Pull request merged**